src/controllers/accounts/AccountsConvertController.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright 2017-2022 Parity Technologies (UK) Ltd.
+// This file is part of Substrate API Sidecar.
+//
+// Substrate API Sidecar is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import { ApiPromise } from '@polkadot/api';
+import { RequestHandler } from 'express';
+import { BadRequest } from 'http-errors';
+
+import { AccountsConvertService } from '../../services/accounts';
+import { IAddressParam, IConvertQueryParams } from '../../types/requests';
+import AbstractController from '../AbstractController';
+
+export default class AccountsConvertController extends AbstractController<AccountsConvertService> {
+	constructor(api: ApiPromise) {
+		super(api, '/accounts/:address/convert', new AccountsConvertService(api));
+		this.initRoutes();
+	}
+
+	protected initRoutes(): void {
+		this.safeMountAsyncGetHandlers([['', this.accountConvert]]);
+	}
+
+	private accountConvert: RequestHandler<
+		IAddressParam,
+		unknown,
+		unknown,
+		IConvertQueryParams
+	> = ({ params: { address }, query: { scheme, prefix, publicKey } }, res) => {
+		// Validation of the `scheme` query param
+		const cryptoScheme = !scheme ? 'sr25519' : scheme;
+		if (
+			!(
+				cryptoScheme === 'ed25519' ||
+				cryptoScheme === 'sr25519' ||
+				cryptoScheme === 'ecdsa'
+			)
+		) {
+			throw new BadRequest(
+				'`scheme` can have one of those 3 values [ed25519, sr25519, ecdsa]'
+			);
+		}
+
+		// Validation of the `prefix` query param
+		const networkPrefix = !prefix ? '42' : prefix;
+		const ss58Prefix = this.parseNumberOrThrow(
+			networkPrefix,
+			'`prefix` provided is not a number.'
+		);
+
+		// Validation of the `publicKey` query param
```

Comment:
This is alright for now, but just for tracking sake, once https://github.com/paritytech/substrate-api-sidecar/pull/1023 goes in, we can update the code to reflect this endpoint.

Reply:
Noted down so I keep an eye when the other PR is merged to come back to this one! Thank you! :pray: